CMakeLists.txt

```diff
@@ -1,32 +1,47 @@
 cmake_minimum_required(VERSION 3.16)
-project(Noxis)
+project(Noxis C)
 
-set(CMAKE_C_FLAGS "-Wall -Wpedantic -Wextra -Werror")
-set(CMAKE_C_STANDARD 11)
+include(cmake/base.cmake)
+include(cmake/warnings.cmake)
 
-set(CMAKE_BINARY_DIR ${PROJECT_SOURCE_DIR})
```

Comment:
CMake is meant for out-of-source builds. Avoid generating artifacts in your source tree. They should be generated inside of the build directory.